yajsapi/runner/index.ts

```diff
@@ -608,20 +608,23 @@ export class Engine {
         await sleep(2);
         if (
           Object.keys(offer_buffer).length > 0 &&
-          workers.size < self._conf.max_workers
+          workers.size < self._conf.max_workers &&
+          work_queue.has_unassigned_items()
         ) {
           let _offer_list = Object.entries(offer_buffer);
           let _sample =
-            _offer_list[
-              Math.floor(Math.random() * Object.keys(offer_buffer).length)
-            ];
+            _offer_list
+              .map(x => { return { obj: x, rnd: Math.random() }; })
+              .sort((a, b) => a.rnd - b.rnd)
+              .map(x => x.obj)
+              .reduce((item, acc) => item[1].score > acc[1].score ? item : acc);
```

Comment:
order of props should be accumulator, item

Reply:
Fixed